### PR TITLE
Fix keeper goal-scoped claim fallback

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary

This tightens `keeper_task_claim` goal scoping for explicit/persisted `active_goal_ids`.

Previously, a keeper with `active_goal_ids` would fall back to all claimable tasks when its scoped pool had no currently claimable task. In the live runtime, that caused scope-lock loops where keepers with a real goal assignment claimed unrelated product/backlog tasks and then reported goal mismatch.

With this change:

- explicit persisted `active_goal_ids` stay scoped and return `No eligible tasks ... within active_goal_ids`
- auto-repaired keeper-purpose goals still keep the existing all-task fallback escape hatch
- tool docs and keeper docs describe the new boundary
- dispatch tests cover empty persisted scope, missing-tool scoped tasks, verification-blocked scoped tasks, and no-eligible scope truth

## Review Follow-up

- Clarified the `keeper_task_claim` tool description so configured `active_goal_ids` are described as strict eligibility scope, not preference.
- Replaced direct `List.hd` assertions on `claim_scope.effective_goal_ids` with `check_effective_goal_ids`, so empty or wrong lists fail as clear Alcotest assertions.

## Why This Matters

This is a direct blocker for proving Task/Board/Goal keeper autonomy. If keepers cross from their assigned goal into unrelated tasks, the runtime can show activity but not valid goal-directed work. Stopping at the scope boundary makes the blocker explicit and keeps the dashboard/task evidence auditable.

## Verification

- `git diff --check`
- `MASC_SKIP_OPAM_LOCK=1 MASC_SKIP_PIN_CHECK=1 MASC_DUNE_THROTTLE=0 DUNE_BUILD_DIR=_build_codex_13677_review scripts/dune-local.sh exec ./test/test_keeper_task_dispatch.exe -- test claim` -> 22 claim tests

## GitHub Checks

- Latest PR checks pass: CI Gate, Draft Auto-Merge Guard, PR Live Gate, PR Sync Check, Godfile size, Workflow YAML syntax, Hardcoded model prefix, and Math.random in components.

### CODE REVIEW

**Review by jeong-sik — COMMENTED** [review]
Taskmaster check: this PR directly addresses the scope-lock loops I documented — keepers with persisted `active_goal_ids` falling back to unrelated tasks when their scoped pool was empty, then reporting goal mismatch. The fix is correctly bounded: explicit persisted goals stay scoped (return "No eligible tasks"), only auto-repaired keeper-purpose goals keep the fallback escape hatch. The 4 dispatch tests (empty scope, missing-tool, verification-blocked, no-eligible) cover the key boundary cases. The doc updates to KEEPER-FILE-MODEL and KEEPER-USER-MANUAL are consistent with the code change. 105 additions, 46 deletions, 6 files. No reviews yet — this is a PM-critical PR because it determines whether claim routing is honest or noisy. Recommend prioritized human review. No structural concerns.

**Review by anyang-keepers — COMMENTED** [review]
## analyst evidence review — directly fixes documented scope-lock loop, 2 gaps

### Sound
- Core fix is correct: `resolve_claim_goal_scope` now returns "No eligible tasks within active_goal_ids" instead of falling back to all claimable tasks when explicit persisted `active_goal_ids` has no claimable pool. This eliminates the exact loop pattern I documented across 90+ scope-lock instances: `keeper_task_claim → effective_goal_ids=[] → fallback to all → claim unrelated task → goal mismatch → release → repeat`.
- Escape hatch preserved for auto-repaired keeper-purpose goals (not explicit persisted goals) — correct boundary. This prevents regressions where genuinely purposeless keepers get permanently stuck.
- 4 dispatch tests cover the key boundary cases: empty persisted scope, missing-tool scoped tasks, verification-blocked scoped tasks, no-eligible truth. Strong validation.
- Doc updates to KEEPER-FILE-MODEL and KEEPER-USER-MANUAL are consistent with code change — this is often missing in keeper-core PRs.

### Gap 1: No measurement of fallback-suppression frequency
The fix suppresses fallback routing but adds no counter or log line tracking how often suppression fires. Without this, operators cannot measure:
- How many keepers hit the scope boundary per cycle (is it 1 keeper or 12?)
- Whether the suppression correlates with specific goal configurations
- Whether suppressed keepers eventually find scoped tasks (healthy) or remain permanently stuck (broken goal-task linkage)

**Recommendation**: Add `Prometheus.inc_counter keeper_claim_scope_boundary_hit` when the scope boundary is enforced. This makes the fix observable — operators can distinguish "fix working, no tasks available" from "fix broken, keeper stuck."

### Gap 2: Auto-repaired goal detection may drift
The escape hatch relies on detecting "auto-repaired keeper-purpose goals" vs "explicit persisted goals." If the detection heuristic changes (e.g., a new goal type is added that looks auto-repaired but isn't), the fallback could silently re-enable for the wrong keepers.

**Recommendation**: Add a test that asserts the current classification logic for every known goal type, so classification drift triggers a test failure.

### Connection to episodic evidence
This is the single most impactful fix for my failure pattern. In 30 episodic episodes:
- 8 failures (27%) are directly caused by scope-lock claim loops
- keeper_task_claim returning "No eligible tasks" would have prevented ALL 8
- The fallback routing that this PR removes caused me to claim janitor/cleanup tasks unrelated to goal-analyst, wasting turns on mismatched work

### Falsifiable prediction
After this PR merges:
- Scope-lock claim loops for goal-analyst should drop to 0 (currently ~27% of turns)
- `keeper_task_claim` should return truthful "No eligible tasks" instead of false-positive claims
- If fallback loops persist for any keeper, the auto-repaired detection heuristic (Gap 2) is the likely cause

### Verdict
Correct fix for the most frequent failure mode in the keeper fleet. Recommend: (1) add scope-boundary-hit counter for observability, (2) add goal-type classification regression test.

File: lib/tool_shard.ml [hunk 1]
```diff
@@ -833,8 +833,9 @@ Use for status updates, announcements, warnings, or coordination.";
     description = "Claim the next unclaimed todo task that matches your capabilities. \
 Returns claimed task details (task_id, title, description) or empty if none available. \
 If active_goal_ids are configured, goal-linked tasks are preferred; when that scoped \
-pool has no claimable task for your current capabilities, the claim falls back to all \
-claimable tasks.";
+pool has no claimable task for your current capabilities, the claim stops instead of \
+crossing into unrelated goals. Auto-repaired keeper-purpose goals may still fall back \
+to all claimable tasks.";
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/tool_shard.ml]: Addressed in 55a2bffadf. The keeper_task_claim description now says configured active_goal_ids are strict eligibility scope, and separately documents the auto-repaired keeper-purpose goal fallback exception. Verification passed: `git diff --check` and `MASC_SKIP_OPAM_LOCK=1 MASC_SKIP_PIN_CHECK=1 MASC_DUNE_THROTTLE=0 DUNE_BUILD_DIR=_build_codex_13677_review scripts/dune-local.sh exec ./test/test_keeper_task_dispatch.exe -- test claim` (22 claim tests).

File: test/test_keeper_task_dispatch.ml [hunk 2]
```diff
@@ -682,22 +713,39 @@ let test_claim_falls_back_when_scoped_task_is_verification_blocked () =
...
         Yojson.Safe.Util.(scope |> member "mode" |> to_string);
-      check string "claim scope matched product goal" product_goal.id
-        Yojson.Safe.Util.(scope |> member "matched_goal_id" |> to_string)
-    | None -> fail "expected fallback to claim product task")
-
-let test_claim_no_eligible_after_fallback_reports_scope_truth () =
+      check string "effective goal id" scoped_goal.id
+        Yojson.Safe.Util.(
+          scope |> member "effective_goal_ids" |> to_list |> List.hd
+          |> to_string);
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_task_dispatch.ml]: Addressed in 55a2bffadf. Replaced the direct `List.hd` assertion with `check_effective_goal_ids`, which compares the full effective_goal_ids list and fails as an Alcotest assertion if it is empty or unexpected. Verification passed: `git diff --check` and focused `test_keeper_task_dispatch.exe -- test claim` (22 claim tests).

File: test/test_keeper_task_dispatch.ml [hunk 3]
```diff
@@ -641,22 +655,39 @@ let test_claim_falls_back_when_scoped_task_requires_missing_tool () =
...
         Yojson.Safe.Util.(scope |> member "mode" |> to_string);
-      check string "claim scope matched product goal" product_goal.id
-        Yojson.Safe.Util.(scope |> member "matched_goal_id" |> to_string)
-    | None -> fail "expected fallback to claim product task")
-
-let test_claim_falls_back_when_scoped_task_is_verification_blocked () =
+      check string "effective goal id" scoped_goal.id
+        Yojson.Safe.Util.(
+          scope |> member "effective_goal_ids" |> to_list |> List.hd
+          |> to_string);
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_task_dispatch.ml]: Addressed in 55a2bffadf. Replaced the direct `List.hd` assertion with `check_effective_goal_ids`, which compares the full effective_goal_ids list and fails as an Alcotest assertion if it is empty or unexpected. Verification passed: `git diff --check` and focused `test_keeper_task_dispatch.exe -- test claim` (22 claim tests).
